src/librustdoc/html/format.rs

```diff
@@ -222,15 +222,18 @@ impl clean::Generics {
         cx: &'a Context<'tcx>,
     ) -> impl fmt::Display + 'a + Captures<'tcx> {
         display_fn(move |f| {
-            let real_params =
-                self.params.iter().filter(|p| !p.is_synthetic_type_param()).collect::<Vec<_>>();
-            if real_params.is_empty() {
+            let mut real_params = self.params.iter().filter(|p| !p.is_synthetic_type_param());
+            let first_param = if let Some(first_param) = real_params.next() {
+                first_param
+            } else {
                 return Ok(());
-            }
+            };
+            let real_params = iter::once(first_param).chain(real_params);
```

Comment:
I think it'd be simpler to just add `.peekable()` onto `self.params.iter().filter(...)` above and then use `if real_params.peek().is_none()`. It would make this change a lot smaller and simplify the code.

Reply:
I did consider that, but personally I prefer not using an extra abstraction (especially one that seems to have a *slight* memory overhead) for just one element. Still, I see your point; making code readable/comprehensible is important. I'll get to it in a bit.